speechbrain/core.py

```diff
@@ -1133,6 +1174,7 @@ def fit_batch(self, batch):
             scaled_loss = self.scaler.scale(
                 loss / self.grad_accumulation_factor
             )
+            self.check_loss_isfinite(scaled_loss)
```

Comment:
Checking for all parameters to be finite is redundant with the GradScaler. It already does this. Also it can be crazy expensive for very large large models. Checking if the loss is not finite makes sense, not the parameters.

Reply:
The grad scaler does this but it does not care for how long the parameters have gone infinite. An idea I suggested yesterday was to occasionally check the gradscaler scale for insane values with a patience mechanism, as I've sometimes seen the scale vanish or explode when issues occurred.  
That does induce a CPU-GPU sync though, but I'm not sure how often we have one in the first place.

Reply:
I agree with @TParcollet. I will remove the part where I'm checking the NaNs/inf in the weights and will only cover the loss part. BTW, the function was also intended for other non-GradScaler use cases like fp32.